ui/src/test/java/edu/wpi/grip/ui/codegeneration/SwitchTest.java

```diff
@@ -0,0 +1,92 @@
+package edu.wpi.grip.ui.codegeneration;
+
+import edu.wpi.grip.core.OperationMetaData;
+import edu.wpi.grip.core.Source;
+import edu.wpi.grip.core.Step;
+import edu.wpi.grip.core.operations.composite.SwitchOperation;
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sources.MockNumberSource;
+import edu.wpi.grip.core.util.ExceptionWitness;
+import edu.wpi.grip.ui.codegeneration.tools.GenType;
+import edu.wpi.grip.ui.codegeneration.tools.PipelineInterfacer;
+
+import org.junit.Test;
+
+import javax.inject.Inject;
+
+
+import static org.junit.Assert.assertEquals;
+
+public class SwitchTest extends AbstractGenerationTest {
+  
+  private int onTrueSourceNum = 0;
+  private int onFalseSourceNum = 1;
+  @Inject
+  ExceptionWitness.Factory ewf;
+
+  boolean setupNum(Number onTrue, Number onFalse, Boolean initVal) {
+    MockNumberSource srcTrue = new MockNumberSource(ewf, onTrue.doubleValue(), osf);
+    MockNumberSource srcFalse = new MockNumberSource(ewf, onFalse.doubleValue(), osf);
+    return setup(srcTrue, srcFalse, initVal);
+  }
+
+  boolean setup(Source onTrue, Source onFalse, Boolean initVal) {
+    Step step = gen.addStep(new OperationMetaData(
+        SwitchOperation.DESCRIPTION, () -> new SwitchOperation(isf, osf)));
+    for (InputSocket sock : step.getInputSockets()) {
+      String sockHint = sock.getSocketHint().getIdentifier();
+      if (sockHint.equalsIgnoreCase("if True")) {
+        gen.connect(onTrue.getOutputSockets().get(0), sock);
+      } else if (sockHint.equalsIgnoreCase("if False")) {
+        gen.connect(onFalse.getOutputSockets().get(0), sock);
+      } else if (sockHint.equals("switch")) {
+        sock.setValue(initVal);
+      }
+    }
+    return true;
```

Comment:
No because then we would need to do a lambda calling that and then return true. There is no java predefined no parameter void return function type.